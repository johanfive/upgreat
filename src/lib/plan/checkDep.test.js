const axios = require('axios')
jest.mock('axios')

const getChangelog = require('./getChangelog')
jest.mock('./getChangelog')

const checkDep = require('./checkDep')

it('should return if unrecognizable semVer', async () => {
  await expect(
    checkDep({
      name: 'some-exotic',
      version: 'file://../../hard/path',
      dev: false,
    }),
  ).resolves.toMatchSnapshot()
})

it('should return error if get npm fails', async () => {
  axios.get.mockImplementationOnce(() =>
    Promise.reject({
      response: {
        status: 404,
      },
    }),
  )

  await expect(
    checkDep({
      name: '404-pkg',
      version: '3.4.5',
    }),
  ).resolves.toMatchSnapshot()
})

it('should return error if package is unpublished', async () => {
  axios.get.mockImplementationOnce(() =>
    Promise.resolve({
      data: {
        time: {
          unpublished: {
            time: 'some date',
          },
        },
      },
    }),
  )

  await expect(
    checkDep({
      name: 'unpublished-pkg',
      version: '3.4.5',
    }),
  ).resolves.toMatchSnapshot()
})

it('should return dependency details if no errors', async () => {
  axios.get.mockImplementationOnce(() =>
    Promise.resolve({
      data: {
        versions: {
          '1.2.3': {},
          '1.2.4': { version: '1.2.4' },
          '2.0.0': { version: '2.0.0' },
        },
        'dist-tags': {
          latest: '2.0.0',
        },
        repository: {
          url: 'repo-url.com',
        },
        time: {},
      },
    }),
  )
  getChangelog.mockImplementationOnce(() =>
    Promise.resolve([
      'http://readme.address.here',
      'http://changelog.address.here',
    ]),
  )

  await expect(
    checkDep({
      name: 'my-pkg',
      version: '^1.2.3',
      dev: true,
    }),
  ).resolves.toMatchSnapshot()
})

it('should return dependency details if no repo url', async () => {
  axios.get.mockImplementationOnce(() =>
    Promise.resolve({
      data: {
        versions: {
          '1.2.3': {},
          '1.2.4': { version: '1.2.4' },
          '2.0.0': { version: '2.0.0' },
        },
        'dist-tags': {
          latest: '2.0.0',
        },
        time: {},
      },
    }),
  )
  getChangelog.mockImplementationOnce(() =>
    Promise.resolve([
      'http://readme.address.here',
      'http://changelog.address.here',
    ]),
  )

  await expect(
    checkDep({
      name: 'my-pkg',
      version: '^1.2.3',
      dev: true,
    }),
  ).resolves.toMatchSnapshot()
})
