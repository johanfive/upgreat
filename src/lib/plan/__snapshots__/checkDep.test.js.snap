// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return dependency details if no errors 1`] = `
Object {
  "changelog": "http://changelog.address.here",
  "dev": true,
  "diff": "major",
  "name": "my-pkg",
  "peers": Array [],
  "repo": "http://readme.address.here",
  "targetVersion": "2.0.0",
  "version": "1.2.4",
}
`;

exports[`should return error if get npm fails 1`] = `
Object {
  "err": 404,
  "name": "404-pkg",
  "peers": Array [],
  "version": "3.4.5",
}
`;

exports[`should return error if package is unpublished 1`] = `
Object {
  "err": "unpublished at some date",
  "name": "unpublished-pkg",
  "peers": Array [],
  "version": "3.4.5",
}
`;

exports[`should return if exotic 1`] = `
Object {
  "err": "exotic",
  "name": "some-exotic",
  "peers": Array [],
  "version": "file://../../hard/path",
}
`;
